lib/spack/spack/directives.py

```diff
@@ -413,24 +413,38 @@ def _execute_conflicts(pkg):
 
 
 @directive(('dependencies'))
-def depends_on(spec, when=None, type=default_deptype, patches=None):
+def depends_on(specs, when=None, type=default_deptype, patches=None, skip=None):
     """Creates a dict of deps with specs defining when they apply.
 
     Args:
-        spec (spack.spec.Spec or str): the package and constraints depended on
+        spec (spack.spec.Spec, str or list): the packages and constraints depended on
         when (spack.spec.Spec or str): when the dependent satisfies this, it has
             the dependency represented by ``spec``
         type (str or tuple): str or tuple of legal Spack deptypes
         patches (typing.Callable or list): single result of ``patch()`` directive, a
             ``str`` to be passed to ``patch``, or a list of these
+        skip (list): packages for which this depends_on call shall do nothing. This
+            is helpful when a base class of other packages like AutotoolsPackage wishes
+            to define dependencies for all subclasses, except for these packages.
+            Example: Add autotools packages for develop builds, but not on themselfes.
```

Comment:
We should probably think more about this `skip` argument. I am wondering if it makes sense to have an operation that is the opposite of `when` for instance:
```python
depends_on('foo', unless='platform=darwin')
```
Another thought is that here we are discarding metadata from the Python code, while we should probably codify it in ASP and let `clingo` decide whether this should be taken into account or not. This just to have a single place where constraints are resolved.